src/GafferSceneUI/StandardLightVisualiser.cpp

```diff
@@ -919,6 +931,101 @@ IECoreGL::ConstRenderablePtr StandardLightVisualiser::quadWireframe()
 	return group;
 }
 
+IECoreGL::ConstRenderablePtr StandardLightVisualiser::quadPortal( const V2f &size )
+{
+	// Portals visualise differently as they only allow light through
+	// their area. Effectively a hole cut in a big plane. We try to
+	// represent this by shading outside of the quad area.
+
+	IntVectorDataPtr vertsPerCurveData = new IntVectorData;
+	V3fVectorDataPtr pData = new V3fVectorData;
+
+	std::vector<int> &vertsPerCurve = vertsPerCurveData->writable();
+	std::vector<V3f> &p = pData->writable();
+
+	// Basic outline of the portal area
+
+	vertsPerCurve.push_back( 4 );
+	p.push_back( V3f( -size.x/2, -size.y/2, 0  ) );
+	p.push_back( V3f( size.x/2, -size.y/2, 0	) );
+	p.push_back( V3f( size.x/2, size.y/2, 0  ) );
+	p.push_back( V3f( -size.x/2, size.y/2, 0	) );
```

Comment:
Odd formatting.

Reply:
It is! https://github.com/GafferHQ/gaffer/pull/3533/commits/93f35d8e86050b362b4a55629a6851a019bdd2d7